backend/src/users/entities/user.entity.ts

```diff
@@ -45,7 +45,13 @@ export class User {
   @Exclude()
   @OneToMany(() => Membership, (membership) => membership.user)
   memberships: Membership[];
+  @Column('int', { array: true, default: [] })
   @Exclude()
   @RelationId((user: User) => user.memberships)
   membershipIds: number[];
+
+  @Column('int', { array: true, default: [] })
+  friendshipsIds: number[];
+  @Column('int', { array: true, default: [] })
+  blocksIds: number[];
```

Comment:
I think it might be worth considering a `ManyToMany` relationship between User and Friendship/Block. This would allow each Friendship/Block to be defined by one row in the Friendship/Block table, but be referenced by both Users involved (rather than manually creating an array to stock the friendship ids). 

It would function similarly to the array, but it would allow us to more easily query what Friendships a User has, and it would avoid storing an array in the database

Reply:
I thought about that, since managing those arrays is really not elegant at all, but doesn't that implies losing the source / target "roles" in the friendship/ block ? I have a hard time warping my head around Many to Many i think..

Reply:
In writing out a description for how that would work, I realized that it probably isn't the best solution because it creates another table in the database that isn't strictly necessary.

It might be simpler to remove that information from the User completely. Then, to see what friends a user has we would query the Friendship Repository for the case where `source == user || target == user`, which would avoid creating a new column or a new table

Reply:
I'll remove it then ! My only concern was efficiency but for the purpose of this project I guess handling it with that query will be good enough.